### ISSUE TITLE
Single-term optimization improvement possibility

### ISSUE THREAD

**Issue by Krzmbrzl** [ValeevGroup/SeQuant#152]
https://doi.org/10.1103/PhysRevE.90.033315 describes some pruning methods to the single-term factorization (aka: the operation minimization of a tensor network), that would probably allow to speed this part of SeQuant's processing up quite a bit.

This might become very relevant in case one ever wants to treat expressions with a very large amount of tensors in a single network or if one wants to step into the direction of some sort of global-ish optimizations of the given expressions (beyond a single tensor network).

**Comment by bimalgaudel** [comment]
Yes, this will be useful for when we have really large tensor networks. For tensor networks with 10 to 12 tensors, the current implementation is fast enough (very fast). Tree pruning might be implemented in the future.

**Comment by bimalgaudel** [comment]
The consensus is, exhaustive search can be used for tensor networks of size up to 18. See section 2 (Previous work) second sentence here:

[Eli Meirom, Haggai Maron, Shie Mannor, Gal Chechik Proceedings of the 39th International Conference on Machine Learning, PMLR 162:15278-15292, 2022.].(https://proceedings.mlr.press/v162/meirom22a.html)
